WebInterface/Index.html

```diff
@@ -147,58 +199,45 @@
         }
     }
 
-    var CategoryTypes = [ "None", "Bar", "Restaurant", "Club" ];
+    function addInfoWindow(marker, barInfo) {
+        var infowindow = new google.maps.InfoWindow({
+            content: createBarDescription(barInfo)
+        });
+        marker.addListener('click', function () {
+            infowindow.open(map, marker);
+        });
+    }
+
+    function createBarDescription(barInfo) {
+        var description = "<p>Place name: <b>" + barInfo.Title + "</b><\p>" +
+                          "<p>Average rating: <b>" + barInfo.AvgRating + "</b><\p>" +
+                          "<p>Categories: <b>" + enumToText(barInfo.Categories, CategoryTypes) + "</b><\p>" +
+                          "<p>Place address: <b>" + barInfo.StreetAddress + "</b><\p>";
+        return description;
+    }
+
+    var CategoryTypes = ["None", "Bar", "Restaurant", "Club"];
```

Comment:
categoryTypes?

Reply:
These are for translating flag enums to normal human words. If we have category enum 5 decimal (101 binary), it's going to be a Bar and a Club. Couldn't find a normal way of making the enum to string so made my own method.